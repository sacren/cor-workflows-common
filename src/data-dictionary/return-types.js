/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */
export const BOOLEAN = 'boolean'
export const CATEGORY = 'category'
export const CATEGORY_LIST = 'category-list'
export const FORM = 'form'
export const GROUP = 'group'
export const GROUP_LIST = 'group-list'
export const NUMBER = 'number'
export const ROLE = 'role'
export const SUBFLOW = 'subflow'
export const TEXT = 'text'
export const TEXT_LIST = 'text-list'
export const USER = 'user'
export const OPTIONS = 'options'
export const ALL = '*'
