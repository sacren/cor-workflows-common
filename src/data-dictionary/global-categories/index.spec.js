/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import GlobalCategory from './'
import { CATEGORY } from '../return-types'
import TestDeflation from '../../test/utils/deflation'

describe('Global category contexts', () => {
  it('should inflate nothing', async () => {
    expect(await GlobalCategory.inflate()).toBeUndefined()
  })

  describe('getChildren', () => {
    it('should throw an error if no context is provided', async () => {
      const groups = new GlobalCategory()
      await expect(groups.getChildren()).rejects.toThrowErrorMatchingSnapshot()
    })

    it('should call the categories api and return a list of category objects', async () => {
      const categories = [{ name: 'category1' }, { name: 'category2' }]
      const ctx = {
        apis: {
          categories: {
            list: jest.fn().mockReturnValue(categories)
          }
        }
      }
      const groups = new GlobalCategory(null, [CATEGORY], null, ctx)
      const children = await groups.getChildren()
      expect(children).toHaveLength(categories.length)
      children.forEach((group, index) => {
        expect(group.parent).toEqual(groups)
        expect(group.returnTypes).toEqual(groups.returnTypes)
        expect(group.data).toMatchObject(categories[index])
        expect(group.ctx).toEqual(groups.ctx)
      })
    })
  })

  TestDeflation(parent => new GlobalCategory(parent, null, null), {
    type: GlobalCategory.type,
    name: GlobalCategory.displayName,
    requiresParent: false
  })
})
