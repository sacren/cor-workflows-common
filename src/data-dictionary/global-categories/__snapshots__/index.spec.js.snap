// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global category contexts getChildren should throw an error if no context is provided 1`] = `"received is not a function"`;
